src/Extensions/Diagnostics.HealthChecks/Internal/HttpEndpointHealthCheck.cs

```diff
@@ -65,5 +66,32 @@ protected override async Task<HealthCheckResult> CheckHealthInternalAsync(Health
 
 			return result;
 		}
+
+		private static HttpRequestMessage CloneRequestMessage(HttpRequestMessage message)
+		{
+			HttpRequestMessage clone = new()
+			{
+				Method = message.Method,
+				RequestUri = message.RequestUri,
+				Content = message.Content,
+				Version = message.Version
+			};
+
+			foreach (KeyValuePair<string, IEnumerable<string>> header in message.Headers)
+			{
+				clone.Headers.TryAddWithoutValidation(header.Key, header.Value);
+			}
+
+#if !NETCOREAPP3_1 && !NETSTANDARD2_0

```

Comment:
Should this be || instead of &&? I am thinking the project can't be compiled for both at the same time but I might miss something here.